tests/unittests/graphTest.cpp

```diff
@@ -1202,3 +1202,68 @@ TEST(Graph, hookTest) {
   ASSERT_TRUE(ph);
   ASSERT_EQ(ph, in);
 }
+
+TEST(Graph, cloneContext) {
+  ExecutionEngine EE;
```

Comment:
Yeah, I picked a Function that was complicated enough to have internal placeholders so the test would cover the broadest case. 

I can definitely do what you want here, but I think it's a worse test for the POV of this change. The reason to make this change is so we can run the same Function concurrently on different Contexts, even if it requires a single context at build time. Removing the Function from the test means makes it pretty artificial. Before I do that, do you think this could be fixed by comments?